### PR TITLE
Bugfix/79 manual merge

### PR DESCRIPTION
**Pull request by cayop**
There are 2 ways for to do a merge of 2 devices.
One is automatic and is used from one snapshot to one device. The core of this merge it's in a Sync class in sync.py file in resource/device directory.
The other is from the web site. This feature was incomplete because the merge was not made correctly and has a bad url for endpoint. 
From the app the user select 2 devices. The first of select is the base device. The second of select is the device than replace some features of base device.
The important points for replace are the components, physical properties and hid.
Next is necessary
* add the tags from the secondary device to base device
* add the actions from secondary device to base device
(This adds include not drop the original tags and actions of base device)

I'm inspired in sync process (automatic merge) for to do this manual merge. The advantage of the manual merge is than you don't need check the tags, actions or components of the secondary device, just move or copy them.

The expected behavior is correctly expressed in the test_merge.py

The problem with the endpoint was than from app there are a POST call like this:
```
/devices/238/merge/id=26
```
but the api expected a request like this:
```
/devices/238/merge/?id=26
```
And next parse this id.
Now the actual request is like this:
```
/devices/238/merge/26

This pull request not work in production until the app is updated correcting this error

### CODE REVIEW

**Comment by slamora** [comment]
Sorry, I don't understand how can I review this PR. Which endpoint should I test and what is the expected behaviour? @cayop could please provide more details about that and the bug that fixes this PR (I don't see any linked issue)? Thanks.

**Review by JNadeu — COMMENTED** [review]
I have reviewed all the changes, and have recreated the test manually and it has worked correctly (without using the front-end). I still think we should have more tests for this functionality, for example, if there are two computers that are compatible by hid (same type) but have different tags assigned we need to check that primary device after the manual merge have two tags assigned.

I have also tested with two snapshots that apparently should not be able to perform a manual merge (different device type) and correctly failed with a status code 422. I think this case should also have a test, and some other very generic case in which the manual merge should not be produced I would also add it.

File: tests/test_merge.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        assert set(pc2.components) == set()
+        assert tag in pc1.tags
+        assert not tag in pc2.tags
+
+        for com in components2:
+            assert com in pc1.components
+
+        for com in components1_excluded:
+            assert not com in pc1.components
+
```
Comment (JNadeu) [inline tests/test_merge.py:60]:
```suggestion

@pytest.mark.mvp
def test_manual_merge_different_computers(app: Devicehub, user: UserClient):
    """ Check if the system does not accept the merging of two totally different computer snapshot """
    snapshot_d1a, _ = user.post(import_snap('real-eee-1001pxd.snapshot.11'), res=m.Snapshot)
    snapshot_d2, _ = user.post(import_snap('real-hp.snapshot.11'), res=m.Snapshot)

    pc1_id = snapshot_d1a['device']['id']
    pc2_id = snapshot_d2['device']['id']

    with app.app_context():
        pc1 = d.Device.query.filter_by(id=pc1_id).one()
        pc2 = d.Device.query.filter_by(id=pc2_id).one()
        n_actions1 = len(pc1.actions)
        n_actions2 = len(pc2.actions)
        action1 = pc1.actions[0]
        action2 = pc2.actions[0]
        assert not action2 in pc1.actions

        tag = Tag(id='tag-d1', owner_id=user.user['id'])
        pc1.tags.add(tag)
        db.session.add(pc1)
        db.session.commit()

        tag = Tag(id='tag-d2', owner_id=user.user['id'])
        pc2.tags.add(tag)
        db.session.add(pc2)
        db.session.commit()

        components1 = [com for com in pc1.components]
        components2 = [com for com in pc2.components]
        components1_excluded = [com for com in pc1.components if not com in components2]
        assert pc1.hid != pc2.hid

        uri = '/devices/%d/merge/%d' % (pc1_id, pc2_id)
        result, _ = user.post({'id': 1}, uri=uri, status=422)

        assert pc1.hid != pc2.hid
        assert result['message'][0] == 'The devices is not the same type.'
```

**Review by JNadeu — COMMENTED** [review]
I don't know if it would be a good idea in this PR seeing that in git it is quite behind the testing branch and many changes have been made, if to make the testing merge first on this branch and then already approve the PR. It's just an observation.

File: tests/test_merge.py (hunk 1 above)
Comment (cayop) [reply tests/test_merge.py:60]: I do a new test one with 2 devices and one tag every one.
The result is the device base with the both tags.

I merged testing into this branch. In theory there must be no problems

**Review by JNadeu — APPROVED** [review]
I see it well with the new tests, in principle from my side we can do the merge to testing branch. With the E2E test we will see if there are any very particular cases that can generate conflicts.

**Review by slamora — APPROVED** [review]

**Pull request merged**